03_analysis/color_by_moiety.py

```diff
@@ -60,51 +75,58 @@ def draw_scatter_moiety(x_data, y_data, all_x_subset, all_y_subset,
     """
     print(f"Number of data points in full scatter plot: {len(x_data)}")
 
-    num_methods = len(x_data)
-
     # set plot limits if specified
     if x_range is not None:
         plt.xlim(x_range[0], x_range[1])
     if y_range is not None:
         plt.ylim(y_range[0], y_range[1])
 
     # set log scaling but use symmetric log for negative values
-    #plt.yscale('symlog')
+    # plt.yscale('symlog')
 
-    if what_for == 'paper':
+    if what_for == "paper":
         fig = plt.gcf()
         fig.set_size_inches(4, 3)
         plt.xlabel(x_label, fontsize=10)
         plt.ylabel(y_label, fontsize=10)
         plt.xticks(fontsize=10)
         plt.yticks(fontsize=10)
-        plt_options = {'s':10, 'alpha':1.0}
-        plt.rc('legend', fontsize=10)
+        plt_options = {"s": 10, "alpha": 1.0}
+        plt.rc("legend", fontsize=10)
 
-    elif what_for == 'talk':
+    elif what_for == "talk":
         fig = plt.gcf()
         fig.set_size_inches(8, 6)
         plt.xlabel(x_label, fontsize=14)
         plt.ylabel(y_label, fontsize=14)
         plt.xticks(fontsize=12)
         plt.yticks(fontsize=12)
-        plt_options = {'s':10, 'alpha':1.0}
-        plt.rc('legend', fontsize=14)
+        plt_options = {"s": 10, "alpha": 1.0}
+        plt.rc("legend", fontsize=14)
 
     # generate the plot with full set
-    plt.scatter(x_data, y_data,
-        c='white', edgecolors='grey', **plt_options)
+    plt.scatter(x_data, y_data, c="white", edgecolors="grey", **plt_options)
 
     # generate the plot with subset(s)
-    subset_colors = ['#01959f', '#614051', '#e96058']  # todo generalize
-    for i, (xs, ys, lab) in enumerate(zip(all_x_subset, all_y_subset, labels_subset)):
-        print(f"Number of data points in subset {i}: {len(xs)}")
-        plt.scatter(xs, ys, label=lab, c=subset_colors[i], zorder=2, **plt_options)
-
-    plt.legend(loc=(0.2, 1.04))
-    plt.savefig(out_file, bbox_inches='tight')
+    subset_colors = ["#01959f", "#614051", "#e96058"]  # todo generalize
+    with open("statistics.dat", "a") as file:
+        file.write(
+            f"{method_label:19s},{'all':9s},{len(x_data):9d},{np.average(x_data[~np.isnan(x_data)]):9.3f},{np.std(x_data[~np.isnan(x_data)]):9.3f},{np.average(y_data[~np.isnan(y_data)]):9.3f},{np.std(y_data[~np.isnan(y_data)]):9.3f}\n"
```

Comment:
I would probably break this up into multiple lines for better readability, or define a variable that contains the really long string beforehand